### ISSUE TITLE
Can't launch rackspace VM under OSX

### ISSUE THREAD

**Issue by somada141** [mitchellh/vagrant-rackspace#135]
I'm using Vagrant 1.7.2 (the latest at the time of writing). I tried using the rackspace plugin with the following Vagrantfile:

```
Vagrant.configure("2") do |config|
  # The box is optional in newer versions of Vagrant
  # config.vm.box = "dummy"

  config.vm.synced_folder ".", "/home/vagrant/vagrant_rackspace"

  config.vm.define :ubuntu do |ubuntu|
    ubuntu.ssh.private_key_path = '~/.ssh/id_rsa'
    ubuntu.vm.provider :rackspace do |rs|
      rs.username = <username_w_quotes>
      rs.admin_password = <password_w_quotes>
      rs.api_key  = <key_w_quotes>
      rs.flavor   = /1 GB Performance/
      rs.image    = /Ubuntu/
      rs.rackspace_region = :iad
      rs.public_key_path = '~/.ssh/id_rsa.pub'
    end
  end
end
```

However, upon running `vagrant up --provider=rackspace` I get the following error:

```
/Users/adam/.vagrant.d/gems/gems/excon-0.44.4/lib/excon/middlewares/expects.rb:6:in `response_call': Expected([200, 203]) <=> Actual(401 Unauthorized) (Excon::Errors::Unauthorized)
excon.error.response
  :body          => "{\"unauthorized\":{\"code\":401,\"message\":\"Username or api key is invalid.\"}}"
  :headers       => {
    "Connection"          => "keep-alive"
    "Content-Type"        => "application/json"
    "Date"                => "Wed, 11 Mar 2015 04:11:31 GMT"
    "Server"              => "nginx"
    "X-NewRelic-App-Data" => "PxQGUF9aDwETVlhSBQgFVUYdFGQHBDcQUQxLA1tMXV1dORYzVBJHNQFUZAQUFVFQVThOFlhaUggXER5jLTU3SxJOCEwIFAQcA1UNWQZVH09VGhUCVFNZUlJXVlFRCFNSUwYAET8="
    "vary"                => "Accept, Accept-Encoding, X-Auth-Token"
  }
  :local_address => "**************"
  :local_port    => *****
  :reason_phrase => "Unauthorized"
  :remote_ip     => "**************"
  :status        => 401
	from /Users/adam/.vagrant.d/gems/gems/excon-0.44.4/lib/excon/middlewares/response_parser.rb:8:in `response_call'
	from /Users/adam/.vagrant.d/gems/gems/excon-0.44.4/lib/excon/connection.rb:372:in `response'
	from /Users/adam/.vagrant.d/gems/gems/excon-0.44.4/lib/excon/connection.rb:236:in `request'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/core/connection.rb:81:in `request'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/service.rb:33:in `request_without_retry'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/requests/identity/create_token.rb:15:in `create_token'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/identity.rb:40:in `authenticate'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/identity.rb:78:in `initialize'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/core/service.rb:115:in `new'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/core/service.rb:115:in `new'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/service.rb:115:in `authenticate_v2'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/service.rb:29:in `authenticate'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/compute_v2.rb:180:in `authenticate'
	from /Users/adam/.vagrant.d/gems/gems/fog-1.28.0/lib/fog/rackspace/compute_v2.rb:154:in `initialize'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/core/service.rb:115:in `new'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/core/service.rb:115:in `new'
	from /Users/adam/.vagrant.d/gems/gems/fog-core-1.29.0/lib/fog/compute.rb:42:in `new'
	from /Users/adam/.vagrant.d/gems/gems/vagrant-rackspace-0.1.10/lib/vagrant-rackspace/action/connect_rackspace.rb:38:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:214:in `action_raw'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:191:in `block in action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:516:in `lock'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:178:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:178:in `action'
	from /Users/adam/.vagrant.d/gems/gems/vagrant-rackspace-0.1.10/lib/vagrant-rackspace/provider.rb:33:in `state'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:480:in `state'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/machine.rb:141:in `initialize'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/vagrantfile.rb:75:in `new'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/vagrantfile.rb:75:in `machine'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:614:in `machine'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/plugin/v2/command.rb:168:in `block in with_target_vms'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/plugin/v2/command.rb:192:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/plugin/v2/command.rb:192:in `block in with_target_vms'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/plugin/v2/command.rb:174:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/plugin/v2/command.rb:174:in `with_target_vms'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/plugins/commands/up/command.rb:74:in `block in execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:277:in `block (2 levels) in batch'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:275:in `tap'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:275:in `block in batch'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:274:in `synchronize'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:274:in `batch'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/plugins/commands/up/command.rb:58:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/cli.rb:42:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:301:in `cli'
	from /opt/vagrant/bin/../embedded/gems/gems/vagrant-1.7.2/bin/vagrant:174:in `<main>'
```

Any ideas as to what I might be doing wrongly? I get a similar error when running `vagrant rackspace flavors`

**Comment by smashwilson** [comment]
That's strange. I'm assuming that you've double-checked your API key? Do those credentials work for anything else?

**Comment by somada141** [comment]
Yeah the credentials are definitely correct (if you can verify that they need to be wrapped in quotes when hardcoded into the Vagrantfile because when I fed them straight-up I got different errors).

Nonetheless the error comes up with `vagrant rackspace flavors` as well (unless that one uses the credentials too?)

Is there another way to check the credentials programmatically (I just copied them from the Rackspace interface)? Everyone else in the group just uses the web interface (I was actually trying to showcase Vagrant and convince them to make the switch)

**Comment by smashwilson** [comment]
> ...if you can verify that they need to be wrapped in quotes when hardcoded into the Vagrantfile...

I can indeed confirm that! The Vagrantfile is Ruby syntax, so your username and API key should both be Ruby strings, delimited with `"..."` or `'...'`.

> Nonetheless the error comes up with vagrant rackspace flavors as well (unless that one uses the credentials too?)

Yes - both commands will use the Rackspace credentials that you pass in the Vagrantfile, so it's not *too* surprising that both are failing for you. In fact, it would be a little mysterious if it was working in one place but not the other :wink:

> Is there another way to check the credentials programmatically (I just copied them from the Rackspace interface)?

Sure, you can try using them directly with `curl` from the command-line:

```bash
export RAX_USERNAME="..."
export RAX_API_KEY="..."

curl -s -X POST https://identity.api.rackspacecloud.com/v1.1/auth \
  -H "Content-Type: application/json" \
  -d '{"credentials": {"username": "'${RAX_USERNAME}'", "key": "'${RAX_API_KEY}'"}}'
```

If that gives you a giant blob of JSON, your API key is good, and it's something with Vagrant or Fog (the library we use to talk to the Rackspace API). If you still see a 401 there instead, there's something wrong with your key, instead.

> Everyone else in the group just uses the web interface (I was actually trying to showcase Vagrant and convince them to make the switch)

A worthy goal indeed.

**Comment by somada141** [comment]
Thanks a lot for your help Ash!!!

Just figured it out. While the credentials were indeed correct and it was driving me mental, I decided to make an additional user just for me (everyone in the team is using the master credentials which I'm aware that its a bad practice). Once I did and used those credentials instead everything went through.

I don't know enough about Rackspace to comment as to whether this is the norm but you might want to verify it and/or mention the possibility on the README.

Cheers!
